community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/NeoStoreFileListing.java

```diff
@@ -66,77 +76,56 @@ public NeoStoreFileListing( File storeDir, LabelScanStore labelScanStore,
     {
         final Collection<ResourceIterator<File>> snapshots = new ArrayList<>();
         for ( IndexImplementation indexProvider : legacyIndexProviders.all() )
         {
             ResourceIterator<File> snapshot = indexProvider.listStoreFiles();
             snapshots.add( snapshot );
-            Iterators.addToCollection( snapshot, files );
+            snapshot.stream().map( toNotAStoreTypeFile ).collect( Collectors.toCollection( () -> files ) );
         }
         // Intentionally don't close the snapshot here, return it for closing by the consumer of
         // the targetFiles list.
         return new MultiResource( snapshots );
     }
 
-    private Resource gatherSchemaIndexFiles(Collection<File> targetFiles) throws IOException
+    private Resource gatherSchemaIndexFiles( Collection<StoreFileMetadata> targetFiles ) throws IOException
     {
         ResourceIterator<File> snapshot = indexingService.snapshotStoreFiles();
-        Iterators.addToCollection(snapshot, targetFiles);
+        snapshot.stream().map( toNotAStoreTypeFile ).collect( Collectors.toCollection( () -> targetFiles ) );
         // Intentionally don't close the snapshot here, return it for closing by the consumer of
         // the targetFiles list.
         return snapshot;
     }
 
-    private Resource gatherLabelScanStoreFiles( Collection<File> targetFiles ) throws IOException
+    private Resource gatherLabelScanStoreFiles( Collection<StoreFileMetadata> targetFiles ) throws IOException
     {
         ResourceIterator<File> snapshot = labelScanStore.snapshotStoreFiles();
-        Iterators.addToCollection(snapshot, targetFiles);
+        snapshot.stream().map( toNotAStoreTypeFile ).collect( Collectors.toCollection( () -> targetFiles ) );
         // Intentionally don't close the snapshot here, return it for closing by the consumer of
         // the targetFiles list.
         return snapshot;
     }
 
-    private void gatherNeoStoreFiles( final Collection<File> targetFiles, boolean includeTransactionLogs )
+    private void gatherNeoStoreFiles( final Collection<StoreFileMetadata> targetFiles )
     {
-        File neostoreFile = null;
-        for ( File dbFile : Objects.requireNonNull( storeDir.listFiles() ) )
-        {
-            String name = dbFile.getName();
-            if ( dbFile.isFile() )
-            {
-                if ( name.equals( MetaDataStore.DEFAULT_NAME ) )
-                {   // Keep it, to add last
-                    neostoreFile = dbFile;
-                }
-                else if ( neoStoreFile( name ) )
-                {
-                    targetFiles.add( dbFile );
-                }
-                else if ( includeTransactionLogs && transactionLogFile( name ) )
-                {
-                    targetFiles.add( dbFile );
-                }
-            }
-        }
-        targetFiles.add( neostoreFile );
```

Comment:
The (now deleted) method used to add the `neostore` file last of all the stores. This is iirc to guard for some cases of a semi-copied store, where the existence of the neostore file would signify that a complete copy of the store had been made. I don't see this behaviour above (and if indeed it is so then there's seemingly no test case for it either).